### PR TITLE
Token Validation

### PR DESCRIPTION
**Pull request by shamantraghav**
# Validates tokens before release #

Fixes [DII-303](https://issues.corp.twilio.com/browse/DII-303)

Test case where both tokens are valid: https://github.com/shamantraghav/twilio-cli-core/runs/5048460195?check_suite_focus=true

Test case where REPO_ACCESS_TOKEN is invalid: https://github.com/shamantraghav/twilio-cli-core/actions/runs/1787870269

Test case where AWS tokens are invalid: 
https://github.com/shamantraghav/twilio-cli-core/actions/runs/1788257300

Testing after integrating with the current release work flow: https://github.com/shamantraghav/twilio-cli-core/actions/runs/1788311085

Test case where the release doesn't get triggered if the tokens are not valid:
https://github.com/shamantraghav/twilio-cli-core/actions/runs/1808375759

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-cli-core/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -12,7 +12,23 @@ on:
         description: 'HomeBrew prerelease'
         default: 'false'
 jobs:
+  token-validation:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Validate REPO_ACCESS_TOKEN 
+        uses: actions/checkout@v2
+        with:
+          repository: 'twilio/twilio-oai'
```
Comment (shamantraghav) [inline .github/workflows/release.yml:21]: Checking out a sample repo(OAI in this case) to validate the token

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -49,7 +66,8 @@ jobs:
         run: make test
   update-api-specs:
     runs-on: ubuntu-latest
-#     needs: [test]
+    needs: [ token-validation ]
```
Comment (kridai) [inline .github/workflows/release.yml:69]: test already needs `token-validation` , and update-api-specs: needs `test`, why are we changing this to `token-validation ` here?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -12,7 +12,23 @@ on:
...
+    steps:
+      - name: Validate REPO_ACCESS_TOKEN 
+        uses: actions/checkout@v2
+        with:
+          repository: 'twilio/twilio-oai'
+          token:  ${{ secrets.REPO_ACCESS_TOKEN }}
+      - name: Validate AWS tokens
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
```
Comment (shamantraghav) [inline .github/workflows/release.yml:26]: These tokens have been added to CLI-core as well

**Comment by kridai** [comment]
I see [Notify Release Failed](https://github.com/shamantraghav/twilio-cli-core/runs/5048698880?check_suite_focus=true#logs) getting triggered for failed token validation. I don't think we should be triggering it in this case. Thoughs?

File: .github/workflows/release.yml (hunk 2 above)
Comment (charan678) [reply .github/workflows/release.yml:69]: +1

Comment (shamantraghav) [reply .github/workflows/release.yml:69]: We're currently disabling the dependency on test. Which is why I added it. Once we enable the dependency on test, we don't need the dependency on token-validation as well. I've changed that now

**Comment by shrutiburman** [comment]
> I see [Notify Release Failed](https://github.com/shamantraghav/twilio-cli-core/runs/5048698880?check_suite_focus=true#logs) getting triggered for failed token validation. I don't think we should be triggering it in this case. Thoughts?

On the same note, would it make sense to make this a separate workflow that we can trigger before triggering the release ?

**Comment by shamantraghav** [comment]
> I see [Notify Release Failed](https://github.com/shamantraghav/twilio-cli-core/runs/5048698880?check_suite_focus=true#logs) getting triggered for failed token validation. I don't think we should be triggering it in this case. Thoughts?

Just saw this. Since token validation fails, the current logic treats this failure as part of the release workflow causing the alert. I'll check on how this can be disabled.

**Comment by shamantraghav** [comment]
> > I see [Notify Release Failed](https://github.com/shamantraghav/twilio-cli-core/runs/5048698880?check_suite_focus=true#logs) getting triggered for failed token validation. I don't think we should be triggering it in this case. Thoughts?
> 
> On the same note, would it make sense to make this a separate workflow that we can trigger before triggering the release ?

Ideally, we wouldn't want this. But this would be the only option if there is no other work around

**Comment by shamantraghav** [comment]
> I see [Notify Release Failed](https://github.com/shamantraghav/twilio-cli-core/runs/5048698880?check_suite_focus=true#logs) getting triggered for failed token validation. I don't think we should be triggering it in this case. Thoughts?

This has been fixed: https://github.com/shamantraghav/twilio-cli-core/actions/runs/1808375759

**Review by kridai — APPROVED** [review]

**Review by shrutiburman — APPROVED** [review]

**Pull request merged**